src/opengl.c

```diff
@@ -889,6 +1237,303 @@ bool glx_blur_dst(session_t *ps, int dx, int dy, int width, int height, float z,
+                                  GLuint texture_unit, GLenum tex_tgt, GLuint texture) {
+	glActiveTexture(GL_TEXTURE0 + texture_unit);
+	glBindTexture(tex_tgt, texture);
+	GLint loc_sampler = glGetUniformLocation(prog, sampler_name);
+	glUniform1i(loc_sampler, (GLint)texture_unit);
+}
+
+bool glx_round_corners_dst0(session_t *ps, struct managed_win *w,
+                            const glx_texture_t *ptex attr_unused, int shader_idx, int dx,
+                            int dy, int width, int height, float z, float cr,
+                            const region_t *reg_tgt attr_unused, glx_blur_cache_t *pbc) {
+
+	assert(shader_idx >= 0 && shader_idx <= 1);
+	assert(ps->psglx->round_passes[0].prog);
+	assert(ps->psglx->round_passes[1].prog);
+	const bool have_scissors = glIsEnabled(GL_SCISSOR_TEST);
+	const bool have_stencil = glIsEnabled(GL_STENCIL_TEST);
+	bool ret = false;
+
+	// log_warn("dxy(%d, %d) wh(%d %d) rwh(%d %d) b(%d), f(%d)",
+	//	dx, dy, width, height, ps->root_width, ps->root_height, w->g.border_width,
+	// w->focused);
+
+	// Calculate copy region size
+	glx_blur_cache_t ibc = {.width = 0, .height = 0};
+	if (!pbc)
+		pbc = &ibc;
+
+	int mdx = dx, mdy = dy, mwidth = width, mheight = height;
+	log_trace("%d, %d, %d, %d", mdx, mdy, mwidth, mheight);
+
+	GLenum tex_tgt = GL_TEXTURE_RECTANGLE;
+	if (ps->psglx->has_texture_non_power_of_two)
+		tex_tgt = GL_TEXTURE_2D;
+
+	// Free textures if size inconsistency discovered
+	if (mwidth != pbc->width || mheight != pbc->height)
+		free_glx_bc_resize(ps, pbc);
+
+	// Generate FBO and textures if needed
+	if (!pbc->textures[0])
+		pbc->textures[0] = glx_gen_texture(tex_tgt, mwidth, mheight);
+	GLuint tex_scr = pbc->textures[0];
+
+	pbc->width = mwidth;
+	pbc->height = mheight;
+
+	if (!tex_scr) {
+		log_error("Failed to allocate texture.");
+		goto glx_round_corners_dst_end;
+	}
+
+	// Read destination pixels into a texture
+	glEnable(tex_tgt);
+	glBindTexture(tex_tgt, tex_scr);
+	glx_copy_region_to_tex(ps, tex_tgt, mdx, mdy, mdx, mdy, mwidth, mheight);
+
+	// Texture scaling factor
+	GLfloat texfac_x = 1.0f, texfac_y = 1.0f;
+	if (tex_tgt == GL_TEXTURE_2D) {
+		texfac_x /= (GLfloat)mwidth;
+		texfac_y /= (GLfloat)mheight;
+	}
+
+	// Paint it back
+	{
+		glDisable(GL_STENCIL_TEST);
+		glDisable(GL_SCISSOR_TEST);
+	}
+
+	{
+		const glx_round_pass_t *ppass = &ps->psglx->round_passes[shader_idx];
+		assert(ppass->prog);
+
+		assert(tex_scr);
+		glBindTexture(tex_tgt, tex_scr);
+
+		glBindFramebuffer(GL_FRAMEBUFFER, 0);
+		glDrawBuffer(GL_BACK);
+		if (have_scissors)
+			glEnable(GL_SCISSOR_TEST);
+		if (have_stencil)
+			glEnable(GL_STENCIL_TEST);
+
+		// Our shader generates a transparent mid section
+		// with opaque corners copied from the background texture
+		// We must use blending to get the window pixels to appear
+		glEnable(GL_BLEND);
+		glBlendFunc(GL_ONE, GL_ONE_MINUS_SRC_ALPHA);
+
+		glTexEnvi(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, GL_REPLACE);
```

Comment:
This is the default if I am not mistaken and we don't change that anywhere.

Reply:
@tryone144 correct, probably left over as I was experimenting with different blend methods.

Reply:
to be fair this is not the only function that does this.